managed/src/main/java/com/yugabyte/yw/models/InstanceType.java

```diff
@@ -44,6 +49,13 @@
     NVME
   }
 
+  @Inject
+  public InstanceType(ConfigFactory config) {
```

Comment:
`Config` not `ConfigFactory`
Like I said use TaskGarbageCollector as example.

Reply:
With Config,  always getting NLP(Null pointers), hence used ConfigFactory instead of.

Reply:
Thats odd. In tests or when running server? May be some ordering issue during binding.
Inject 
`private final Provider<Config> config`
@Inject
InstanceType(Provider<Config> config) {
...
}`

then where you use it do 

`config.get()`

This delays the resolution to runtime.

Let me know if that fixes the problem. If not will patch and see whats wrong.

The config factory is not an option here. That code is buggy. 
Another way is to get to typesafe config through play but lets try this first.

Reply:
Its happening while running server. 
Getting below error:
Caused by: com.google.inject.CreationException: Unable to create injector, see the following errors:
1) Error injecting constructor, java.lang.NullPointerException
  at AppInit.<init>(AppInit.java:47)
  at Module.configure(Module.java:48) (via modules: com.google.inject.util.Modules$OverrideModule -> Module)
  while locating AppInit
  
  Tried above things as well but that didn't work out. 
  Updated a patch, using config only. Please suggest! Thanks!

Reply:
So I saw a `provider` for `Config` in `Module`. But looks like that is a different Config type not the `typesafe.Config`. 
So it is not going to work this way. Provider trick also does not delay the resolution because we need it during AppInit.

So way to do this gracefully is 
 - Inject `Application` in `InstanceType`
  - Then you can get to typesafe config using `app.config()`  (NOT `app.configuration()` which is deprecated)
   `this.config = app.config()`
   
   Let me know if that helps.

Reply:
Just noticed that InstanceType is a Entity bean. So it wont use guice injection. It is using the default ctor (that you added) followed by the field mapping by the ORM  and thus config remains null.
So you can just pass the config value into findByProvider as parameter. No config field.

Sorry for confusion. Right way to do this is not use so much static factory methods. Usually there is a singleton helper class like InstanceTypeHelper. That will have non-static findByProvider method. then you can actually do proper guice injection in such class.

Reply:
Done. Added this for now, later we will refactor the code once we have Helper method in place as you suggested above. Thanks!